### PR TITLE
Chore/upgrade to use python 3.12 and pandas 2

### PR DESCRIPTION
**Pull request by DawidMakar-acaisoft**
- Unify the empty value representation to pd.NA
- Update code to support Python 3.12 and Pandas 2.x
- Fix a test for checking UAEUInformation initialization, add a missing test case

### CODE REVIEW

File: opcua_tools/ua_data_types.py [hunk 1]
```diff
@@ -279,44 +281,47 @@ def xml_encode(self, include_xmlns: bool) -> str:
...
+            pd.NA: None,
             float("inf"): '"Infinity"',
             float("-inf"): '"-Infinity"',
             float("nan"): "NaN",
         }
         if self.value in special_floats_or_decimals:
             return special_floats_or_decimals[self.value]
         elif math.isnan(self.value):
-            return '"NaN"'
+            return None
```
[the remark this answers is not in the document]
Comment (DawidMakar-acaisoft) [reply opcua_tools/ua_data_types.py:324]: I remember I changed '"NaN" to None before we were talking about it, and I forgot to switch it back. So I'll do it now.

File: opcua_tools/ua_data_types.py [hunk 2]
```diff
@@ -898,51 +915,52 @@ def json_encode(self, **kwargs) -> str:
...
+        if not isinstance(self.description, UALocalizedText) and isinstance(
+            self.description, str
         ):
-            object.__setattr__(self, "description", UALocalizedText(text=description))
-        elif isinstance(description, UALocalizedText) or display_name is None:
-            object.__setattr__(self, "description", description)
+            object.__setattr__(
+                self, "description", UALocalizedText(text=self.description)
+            )
+        # elif isinstance(self.description, UALocalizedText) or self.display_name is None:
```
[the remark this answers is not in the document]
Comment (DawidMakar-acaisoft) [reply opcua_tools/ua_data_types.py:951]: Yes, this should be removed. Good catch! 👏

File: tests/expected/paper_example/nodes.csv [hunk 3]
```diff
@@ -17,18 +17,18 @@ UAVariable,CA_Y,Normal function output,,ns=1;i=42,CA_Y,,,,,,,3,,3,1,,i=11,
...
+UAVariable,EngineeringUnits,,"UAEngineeringUnits(display_name=UALocalizedText(text='%', locale='en'), description=UALocalizedText(text='percent', locale='en'), unit_id=20529, namespace_uri='http://www.opcfoundation.org/UA/units/un/cefact')",ns=1;i=35,EngineeringUnits,,,,,,,3,,0,1,,i=887,
+UAVariable,EngineeringUnits,,"UAEngineeringUnits(display_name=UALocalizedText(text='%', locale='en'), description=UALocalizedText(text='percent', locale='en'), unit_id=20529, namespace_uri='http://www.opcfoundation.org/UA/units/un/cefact')",ns=1;i=38,EngineeringUnits,,,,,,,3,,0,1,,i=887,
+UAVariable,EngineeringUnits,,"UAEngineeringUnits(display_name=UALocalizedText(text='%', locale='en'), description=UALocalizedText(text='percent', locale='en'), unit_id=20529, namespace_uri='http://www.opcfoundation.org/UA/units/un/cefact')",ns=1;i=41,EngineeringUnits,,,,,,,3,,0,1,,i=887,
+UAVariable,EngineeringUnits,,"UAEngineeringUnits(display_name=UALocalizedText(text='%', locale='en'), description=UALocalizedText(text='percent', locale='en'), unit_id=20529, namespace_uri='http://www.opcfoundation.org/UA/units/un/cefact')",ns=1;i=44,EngineeringUnits,,,,,,,3,,0,1,,i=887,
```
[the remark this answers is not in the document]
Comment (olavpe) [reply tests/expected/paper_example/nodes.csv:31]: Fantastic!! Thank you for adding the info here:))

**Review by olavpe — APPROVED** [review]
Looks good! Great Job Dawid:))

**Pull request merged**